nshm_logic_tree_utilities/lib/logic_tree_tools.py

```diff
@@ -0,0 +1,873 @@
+) -> LogicTree:
+    """
+    Modifies a logic tree to only include branch sets that correspond to the selected tectonic region types.
+
+    Parameters
+    ----------
+    logic_tree : LogicTree
+        The logic tree to modify.
+
+    tectonic_region_type_group : list[param_options.TectonicRegionTypeName] or param_options.TectonicRegionTypeName
+        The selected tectonic region types.
+
+    which_interface : param_options.InterfaceName, default = param_options.InterfaceName.HIK_and_PUY
+        The subduction interfaces to include.
+
+    Returns
+    -------
+    modified_logic_tree : LogicTree
+        The modified logic tree that only includes branch sets corresponding
+        to the selected tectonic region type.
+    """
+
+    if isinstance(tectonic_region_type_group, str):
+        tectonic_region_type_group = [tectonic_region_type_group]
+
+    modified_logic_tree = copy.deepcopy(logic_tree)
+    new_branch_sets = []
+    for branch_set in logic_tree.branch_sets:
+        if isinstance(logic_tree, SourceLogicTree):
+
+            ## even though each branch_set corresponds to one tectonic region type,
+            ## branch_set.tectonic_region_types returns a list of one tectonic region
+            # type which is accessed with the for loop
+            for tectonic_region_type in branch_set.tectonic_region_types:
+                if tectonic_region_type in tectonic_region_type_group:
+                    if (
+                        tectonic_region_type
+                        == param_options.TectonicRegionTypeName.Subduction_Interface
+                    ):
+                        if which_interface == param_options.InterfaceName.HIK_and_PUY:
+                            new_branch_sets.append(copy.deepcopy(branch_set))
+                        elif which_interface == param_options.InterfaceName.only_HIK:
+                            if branch_set.short_name == "HIK":
+                                new_branch_sets.append(copy.deepcopy(branch_set))
+                        elif which_interface == param_options.InterfaceName.only_PUY:
+                            if branch_set.short_name == "PUY":
+                                new_branch_sets.append(copy.deepcopy(branch_set))
+                    else:
+                        new_branch_sets.append(copy.deepcopy(branch_set))
+
+        if isinstance(logic_tree, GMCMLogicTree):
+            if branch_set.tectonic_region_type in tectonic_region_type_group:
+                new_branch_sets.append(copy.deepcopy(branch_set))
+
+    modified_logic_tree.branch_sets = new_branch_sets
+    branch_set_short_names = [x.short_name for x in new_branch_sets]
+
+    if (param_options.InterfaceName.only_PUY in branch_set_short_names) & (
+        param_options.InterfaceName.only_HIK in branch_set_short_names
+    ):
+        # retain the HIK to PUY correlations
+        pass
+    else:
+        # remove correlations
+        modified_logic_tree.correlations = LogicTreeCorrelations()
```

Comment:
branch_set_short_names is just a list like ["CRU", "HIK", "PUY"] that comes from the GNS code so this can be as simple as:
    `if not (("HIK" in branch_set_short_names) and ("PUY" in branch_set_short_names)):`
    `    # HIK and PUY are not both in the logic tree so remove their correlations`
   `    modified_logic_tree.correlations = LogicTreeCorrelations()`